src/matlab/LRNURBSSurface.m

```diff
@@ -378,13 +378,19 @@ function raiseOrder(this, dp, dq)
 			sup = this.support{iel};
 			w   = this.cp(end,sup); % weight controlpoints
 			W   = z(end,1);         % weight function
-			if(derivs > -1)
+            if(derivs > -1)
 				N(1,:) = N(1,:).*w / W;
-			end
-			if(derivs > 0)
+            end
+            if(derivs > 0)
 				N(2,:) = (N(2,:)*W-N(1,:)*z(end,2)).* w / W^2;
 				N(3,:) = (N(3,:)*W-N(1,:)*z(end,3)).* w / W^2;
-			end
+            end
+            if(derivs > 1)
```

Comment:
Does it even get to this point in the code? Will not line 371 catch this case and stop execution?

Reply:
I'll make a new commit changing the condition at line 371 + using tabs and not spaces + minor changes